### PR TITLE
Issue#11872 Resolved

### PR DESCRIPTION
**Pull request by 007durgesh219**

### CODE REVIEW

**Comment by ibennetch** [comment]
You currently have three commits, could you squash them to make one? In the process, it would be helpful to change your commit message to something like "Fix HTML rendering in error message, issue 11873" or something like that. You can force push your amended commits back to the same branch which will update this pull request, there's no need to create a new one.

**Comment by 007durgesh219** [comment]
Thanks @ibennetch I have done require changes.

**Comment by ibennetch** [comment]
Thanks, that looks great. Someone should complete a final review in the next few days.

By the way, normally a fix like this would target the QA branch (see https://wiki.phpmyadmin.net/pma/Release_Targets), which in this case would be QA_4_5.

**Comment by 007durgesh219** [comment]
So, Do I have to change the target or its fine

**Comment by nijel** [comment]
The fix is wrong, as the error message has to be escaped in some cases (eg. when coming from MySQL server or from URL parameters).

**Comment by 007durgesh219** [comment]
I've corrected it. Please review again.

**Comment by nijel** [comment]
Can't it be done rather at time generating the error message? Regexp parsing of HTML will break at some point. (in this case what if $param contains it).

**Comment by 007durgesh219** [comment]
Yup you are right.  I will generate bbcode at time of generating error then.  It will execlude $param from it.

**Comment by 007durgesh219** [comment]
Should I make a different function( let's say ```showDocuBBCode('faq', 'faqmissingparameters')```  ) for generating bbcode of HTML to make things clean?? In that case I will use ```showDocuBBCode('faq', 'faqmissingparameters')``` insted of ```showDocu('faq', 'faqmissingparameters')``` while generating error message.

**Comment by nijel** [comment]
I think passing it parameter indicating output type (html/bbcode, defaulting to html) would be better.

**Comment by 007durgesh219** [comment]
OK, will do that.

**Comment by 007durgesh219** [comment]
I've done require changes

**Comment by 007durgesh219** [comment]
@nijel can you please review it again?

File: libraries/Util.php [hunk 1]
```diff
@@ -549,16 +549,47 @@ public static function getDocuLink($page, $anchor = '')
     /**
      * Displays a link to the phpMyAdmin documentation
      *
-     * @param string $page   Page in documentation
-     * @param string $anchor Optional anchor in page
+     * @param string  $page   Page in documentation
+     * @param string  $anchor Optional anchor in page
+     * @param boolean $bbcode Optional flag to output bbcode
```
Comment (ibennetch) [inline libraries/Util.php]: Can I suggest making this comment "Optional flag whether to output bbcode"? -- it's a small thing that IMHO makes it more readable.

Comment (ibennetch) [inline libraries/Util.php]: Can I suggest making this comment "Optional flag indicating whether to output bbcode"? -- it's a small thing that IMHO makes it more readable.

**Comment by 007durgesh219** [comment]
Thanks @ibennetch I've changed the message.

File: libraries/Util.php [hunk 2]
```diff
@@ -549,16 +549,47 @@ public static function getDocuLink($page, $anchor = '')
...
      *
      * @return string  the html link
      *
      * @access  public
      */
-    public static function showDocu($page, $anchor = '')
+    public static function showDocu($page, $anchor = '', $bbcode = false)
     {
-        return self::showDocLink(self::getDocuLink($page, $anchor));
+        $doc_link = self::showDocLink(self::getDocuLink($page, $anchor));
```
Comment (nijel) [inline libraries/Util.php]: How about passing bbcode further down, so that we directly generate it instead of doing preg_replace magic?

Comment (007durgesh219) [inline libraries/Util.php]: You mean generating bbcode in ```showDocLink``` instead of ```showDocu```.

Comment (nijel) [inline libraries/Util.php]: Yes - generating the code directly instead of doing preg_replace on it...

Comment (007durgesh219) [inline libraries/Util.php]: Thanks I've done that. Please review it again

**Comment by nijel** [comment]
Thanks, now it looks good! Merged to QA_4_6 and master.

**Pull request merged**